gen/gen.go

```diff
@@ -135,7 +143,23 @@ func (g *Gen) Build(config *Config) error {
 	if err := p.ParseAPIMultiSearchDir(searchDirs, config.MainAPIFile, config.ParseDepth); err != nil {
 		return err
 	}
-	swagger := p.GetSwagger()
+
+	return g.Generate(p.GetSwagger(), &GenConfig{
+		OutputDir:     config.OutputDir,
+		InstanceName:  config.InstanceName,
+		GeneratedTime: config.GeneratedTime,
+	})
+}
+
+// Generate outputs a swagger spec
+func (g *Gen) Generate(swagger *spec.Swagger, config *GenConfig) error {
```

Comment:
I moved out of the Build function the parts that deal with generating the files that store the generated swagger spec, because in my case I don't generate the specs from the swag binary, but I want to create the same files that swag does.